### PR TITLE
Ratings (assignment 42)

### PR DESCRIPTION
**Pull request by dogweather**
> Create a Rating model with the attribute "severity" which can be one of "PG", "PG-13", or "R". . . 

https://www.bloc.io/users/brenda-schmidt/checkpoints/1648?roadmap_section_id=120

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Post < ActiveRecord::Base
   has_many :comments, dependent: :destroy
   has_many :labelings, as: :labelable
   has_many :labels, through: :labelings
+  has_many :ratings, as: :rateable
```
Comment (dogweather) [inline app/models/post.rb]: Look back at the [assignment](https://www.bloc.io/users/brenda-schmidt/checkpoints/1648?roadmap_section_id=120). A Rating has many Posts, not the other way around. Sorry!

File: app/models/rating.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Rating < ActiveRecord::Base
+  has_many :topics
+  has_many :posts
+
+  def self.update_rating(rating_string)
+    return Rating.none if rating_string.blank?
+
+  enum severity: [ :PG, :PG13, :R ]
```
Comment (dogweather) [inline app/models/rating.rb]: This is the out-of-place `enum` that Levi mentioned. The assignment shows where to place it.

File: app/models/topic.rb [hunk 3]
```diff
@@ -2,4 +2,5 @@ class Topic < ActiveRecord::Base
   has_many :posts, dependent: :destroy
   has_many :labelings, as: :labelable
   has_many :labels, through: :labelings
+  has_many :ratings, as: :rateable
```
Comment (dogweather) [inline app/models/topic.rb]: Here too, a topic just has one rating.

File: db/migrate/20151121215520_create_ratings.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class CreateRatings < ActiveRecord::Migration
+  def change
+    create_table :ratings do |t|
+      t.references :rating, index: true
+
+      t.references: :rateable, polymorphic: true, index: true
```
Comment (dogweather) [inline db/migrate/20151121215520_create_ratings.rb:6]: This migration looks like it has syntax errors — it didn't run, did it?

**Comment by dogweather** [comment]
Hi, Brenda. I think we should get Assignment 42 finished up. That'll create a good base for doing the next assignments.

I noted a few problems, above. As far as the wonky assignment text goes that Levi wants to edit: for now, completely skip the `update_rating` part, and also skip this instruction: 

> Lastly, add create and edit functionality for Rating to Bloccit.

File: db/migrate/20151121215520_create_ratings.rb (hunk 4 above)
Comment (bpschmidt) [inline db/migrate/20151121215520_create_ratings.rb:6]: I'm not sure what I did here (did the old assignment maybe suggest doing this?), so I tried to re-run `rails generate model Rating severity:integer`, but it says that this old migration conflicts with the rails generate model. It gave the option to `--force` in the terminal, but I'm not sure what that will do/how to do it

**Comment by dogweather** [comment]
Great. I'll "close" the Pull Request without merging it: that just ends this conversation-space. But we can leave the branch alone, in case we want to come back to it later.

**Pull request closed without merging**